Implement keep-alive functionality

FairPlay137 opened the pull request:
This adds default behavior to send a FLAP keep-alive packet every two minutes (customizable through TOCClientSettings) to prevent the socket from closing. This behavior can be disabled by passing `client.ConnectAsync(false);` in the main program.

FairPlay137 committed:
Implement keepalive loop

FairPlay137 committed:
Add keep-alive interval minimum cap

Animadoria commented on TOCSharp/TOCClientSettings.cs:
```diff
@@ -6,5 +6,6 @@ public class TOCClientSettings
         public ushort Port { get; set; } = FLAPConnection.DEFAULT_PORT;
         public string ClientName { get; set; } = "TOCSharp";
         public bool DebugMode { get; set; } = false;
+        public uint KeepAliveInterval { get; set; } = 120000;
```
Use TimeSpan for the interval, makes it easier to pass on Task.Delay

FairPlay137 replied:
Addressed in latest commit.

Animadoria commented on TOCSharp/TOCClient.cs:
```diff
@@ -195,6 +200,18 @@ private async Task SendCommandAsync(params string[] args)
             });
         }
 
+        private async Task SendKeepAliveAsync()
```
KeepAlive methods should be moved to the FLAP connection class as that's a connection thing

FairPlay137 replied:
The SendKeepAliveAsync task has been moved. If the actual loop itself needs to be moved too, let me know.

Animadoria commented on TOCSharp/TOCClient.cs:
```diff
@@ -18,6 +18,8 @@ public partial class TOCClient
         private FLAPConnection? connection;
         private readonly TOCClientSettings settings;
 
+        private bool isKeepAliveLoopRunning = false;
```
Change variable name to "keepAliveLoopRunning", no need for "is" considering the variable is a boolean

FairPlay137 replied:
Addressed in latest commit.

FairPlay137 committed:
Move SendKeepAliveAsync to FLAPConnection, ...

change KeepAliveInterval to a TimeSpan,
and rename isKeepAliveLoopRunning to just keepAliveLoopRunning
